projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -78,7 +78,17 @@ object R2dbcOffsetStore {
               acc.byPid.updated(r.pid, r)
           }
         val newSeen =
-          if (newByPid eq acc.byPid) acc.seen else acc.seen.removed(r.pid)
+          if (newByPid eq acc.byPid)
+            acc.seen
+          else {
+            acc.seen.filter { case (seenPid, seenSeqNr) =>
+              newByPid.get(seenPid) match {
+                case Some(r) => r.seqNr < seenSeqNr
+                case None    => true
+              }
+            }
```

Comment:
I will move the transient seen state out to a separate AtomicRef because it has different concurrency properties than the state (must allow for concurrent updates of seen)